### PR TITLE
fix bug demonstrations/tutorial_expressivity_fourier_series.py

### PR DESCRIPTION
**Pull request by emgilfuster**
I tried to run the demo on my machine and line 208:
`exponent = np.complex(0, scaling * (idx+1) * x)`
resulted in:
`AttributeError: module 'pennylane.numpy' has no attribute 'complex'`.

Replacing `np.complex` with just `complex` solved the issue.

### CODE REVIEW

**Comment by mariaschuld** [comment]
@antalszava I asked Elies to open this in case the error is due to changes in master and would bug you during the next release... Or do you think it is a local problem?

**Comment by antalszava** [comment]
Sorry about the delayed response here. Locally with the `master` branch, I don't seem to be able to reproduce this
```python
from pennylane import numpy as np

some_imag = 0.34
np.complex(0, some_imag)
```
Which PennyLane version produced the error (e.g., can be checked with `qml.about()`)? It's odd that the issue arises only with `np.complex` and not with other wrapped functions :thinking:

**Comment by emgilfuster** [comment]
Name: PennyLane
Version: 0.16.0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author: None
Author-email: None
License: Apache License 2.0
Location: /home/emgilfuster/venvs/PL/lib/python3.8/site-packages
Requires: semantic-version, appdirs, autograd, autoray, networkx, scipy, numpy, toml
Required-by: 
Platform info:           Linux-5.4.72-microsoft-standard-WSL2-x86_64-with-glibc2.29
Python version:          3.8.5
Numpy version:           1.20.3
Scipy version:           1.6.3

**Comment by antalszava** [comment]
Thank you! It seems `NumPy` the culprit, managed to get a warning out when using vanilla NumPy with version 1.20:
```
DeprecationWarning: `np.complex` is a deprecated alias for the builtin `complex`. To silence this warning, use `complex` by itself. Doing this will not modify any behavior and is safe.

If you specifically wanted the numpy scalar type, use `np.complex128` here.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
```

**Review by antalszava — APPROVED** [review]
Thank you!

File: demonstrations/tutorial_expressivity_fourier_series.py [hunk 1]
```diff
@@ -205,7 +205,7 @@ def target_function(x):
     """Generate a truncated Fourier series, where the data gets re-scaled."""
     res = coeff0
     for idx, coeff in enumerate(coeffs):
-        exponent = np.complex(0, scaling * (idx+1) * x)
+        exponent = complex(0, scaling * (idx+1) * x)
```
Comment (antalszava) [inline demonstrations/tutorial_expressivity_fourier_series.py:208]:
```suggestion
        exponent = np.complex128(0, scaling * (idx+1) * x)
```

**Comment by antalszava** [comment]
@emgilfuster changed to using `np.complex128` which should be an identical fix. This is just for convenience with the PennyLane code bases, we might slightly prefer using `np.complex64` and `np.complex128`.

**Comment by emgilfuster** [comment]
good!

File: demonstrations/tutorial_expressivity_fourier_series.py [hunk 2]
```diff
@@ -205,7 +205,7 @@ def target_function(x):
     """Generate a truncated Fourier series, where the data gets re-scaled."""
     res = coeff0
     for idx, coeff in enumerate(coeffs):
-        exponent = np.complex(0, scaling * (idx+1) * x)
+        exponent = np.complex128(0, scaling * (idx+1) * x)
```
Comment (antalszava) [inline demonstrations/tutorial_expressivity_fourier_series.py:208]: Update: `np.complex128` takes one argument only.
```suggestion
        exponent = np.complex128(scaling * (idx+1) * x * 1j)
```

**Comment by antalszava** [comment]
[ch7634]

**Pull request merged**